### PR TITLE
applications: nrf5340_audio: Moved contin_array

### PR DESCRIPTION
**Pull request by koffes**
Tests and code moved

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@koffes , added a PR that adds missing docs that points at your branch, but I need info from you: https://github.com/koffes/sdk-nrf/pull/2
Can you hold off with merging this PR of yours until my PR is done and merged?

**Review by greg-fer — COMMENTED** [review]
The following docs need to be updated:

- Configuring section on the library documentation page
- Changelog with an entry about the new library

**Review by alexsven — APPROVED** [review]

**Review by rick1082 — APPROVED** [review]

File: include/contin_array.h
Comment (greg-fer) [inline include/contin_array.h:38]: Please add the following lines before the closing #endif to solve the doc build issue:
```
/**
 * @}
 */
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/others/contin_array.rst
Comment (greg-fer) [inline doc/nrf/libraries/others/contin_array.rst:27]: @koffes , do we still need update to this section or the Kconfig was aligned?

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/others/contin_array.rst
Comment (koffes) [reply doc/nrf/libraries/others/contin_array.rst:27]: Awaiting response from reviewers on this issue.

**Review by desowin — APPROVED** [review]

File: include/contin_array.h
Comment (erikrobstad) [inline include/contin_array.h:41]: Remove if "@defgroup" is removed?

File: doc/nrf/libraries/others/contin_array.rst
Comment (erikrobstad) [inline doc/nrf/libraries/others/contin_array.rst:10]:
```suggestion
The continuous array library introduces an array that you can loop over, for example if you want to create a period of a Pulse-code modulated (PCM) sine wave.
```

File: include/contin_array.h
Comment (greg-fer) [reply include/contin_array.h:15]: This is needed for doxygen API docs. Do not remove this.

File: include/contin_array.h
Comment (greg-fer) [reply include/contin_array.h:41]: This is needed for doxygen API docs. Do not remove this.

File: doc/nrf/libraries/others/contin_array.rst
Comment (greg-fer) [reply doc/nrf/libraries/others/contin_array.rst:19]: I understand that this part is final, @koffes ?

**Review by greg-fer — APPROVED** [review]
Approving, but please squash my commit and let me know about the Configuration part.

**Pull request merged**
